models/employees/employee_preferences/day_preference.py

```diff
@@ -1,15 +1,20 @@
 import datetime
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 
 from models.employees.employee_preferences.shifts_preference import ShiftsPreferences
 from models.shifts.shift import Shift
 
 
 @dataclass
 class DayOffPreference(ShiftsPreferences):
-    day_date: datetime.date
+    day_date: list[datetime.date] = field(default_factory=list)
 
     def get_shifts_preferences(self, shifts: list[Shift]) -> list[Shift]:
-        shifts_in_day_off_preference = [shift for shift in shifts if shift.start_time.date() == self.day_date]
+        shifts_in_day_offs_preference = []
```

Comment:
i think it would be better if each preference would get its own "DatePreference" object. where preferences in the future could have names. ie

dont want to work weekends: "DayPref...."

so having a list of datetime.date makes little sense. it should probably just be a range